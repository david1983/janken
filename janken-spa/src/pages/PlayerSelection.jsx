import React from 'react';
import PropTypes from 'prop-types';
import { observer, inject } from 'mobx-react';
import {
  Card, CardContent, CardActions, Button,
} from '@material-ui/core';
import PlayerNameInput from '../components/PlayerNameInput';
import GameClass from '../state/Game';
// import GameClass from '../state/Game';

const PlayerSelection = ({ history, Game }) => (
  <div className="player-selection-container">
    <form onSubmit={(e) => {
      e.preventDefault();
      Game.start();
      history.push('/game');
    }}
    >
      <Card className="player-selection-form">
        <CardContent>
          <h1>Player selection</h1>
          <PlayerNameInput key="player1" playerNumber={1} />
          <PlayerNameInput key="player2" playerNumber={2} />
        </CardContent>
        <CardActions className="action-area">
          <Button
            color="primary"
            variant="contained"
            type="submit"
            className="player-selection-form-submit"
          >
            Start Game
          </Button>

        </CardActions>
      </Card>
    </form>

  </div>
);


PlayerSelection.defaultProps = {
  history: {},
  Game: new GameClass(),
};

PlayerSelection.propTypes = {
  history: PropTypes.object,
  Game: PropTypes.object,
};


export default inject('Game')(observer(PlayerSelection));
