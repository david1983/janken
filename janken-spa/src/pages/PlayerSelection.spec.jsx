import React from 'react';
import { shallow } from 'enzyme';
import PlayerSelection from './PlayerSelection';

describe('the GamePage component', () => {
  it('should match the snapshot of the shallow rendered component', () => {
    const component = <PlayerSelection />;
    const rendered = shallow(component);
    expect(rendered).toMatchSnapshot();
  });
});
