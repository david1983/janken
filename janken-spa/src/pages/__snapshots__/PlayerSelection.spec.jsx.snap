// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`the GamePage component should match the snapshot of the shallow rendered component 1`] = `
<o
  forwardRef={null}
/>
`;
